packages/vite/src/runtime/sourcemap/decoder.ts

```diff
@@ -0,0 +1,63 @@
+import type {
+  InvalidOriginalMapping,
+  OriginalMapping,
+} from '@jridgewell/trace-mapping'
+import { originalPositionFor } from '@jridgewell/trace-mapping'
+import { posixResolve } from '../utils'
+
+interface SourceMapLike {
+  version: number
+  mappings?: string
+  names?: string[]
+  sources?: string[]
+  sourcesContent?: string[]
+}
+
+type Needle = {
+  line: number
+  column: number
+}
+
+export class DecodedMap {
+  _encoded: string
+  _decoded: undefined | number[][][]
+  _decodedMemo: Stats
+  url: string
+  version: number
+  names: string[] = []
+  resolvedSources: string[]
+
+  constructor(
+    public map: SourceMapLike,
+    from: string,
+  ) {
+    const { mappings, names, sources } = map
+    this.version = map.version
+    this.names = names || []
+    this._encoded = mappings || ''
+    this._decodedMemo = memoizedState()
+    this.url = from
+    this.resolvedSources = (sources || []).map((s) =>
+      posixResolve(s || '', from),
+    )
+  }
+}
+
+interface Stats {
+  lastKey: number
+  lastNeedle: number
+  lastIndex: number
+}
+function memoizedState(): Stats {
+  return {
+    lastKey: -1,
+    lastNeedle: -1,
+    lastIndex: -1,
+  }
+}
+export function getOriginalPosition(
+  map: DecodedMap,
+  needle: Needle,
+): OriginalMapping | InvalidOriginalMapping {
+  return originalPositionFor(map as any, needle)
```

Comment:
By the way, the function was modified in more than one way - particularly it returned `null` instead of invalid mapping.

Reply:
I remembered about that while extracting the `@jridgewell/trace-mapping` but later forgot. 🫠

Reply:
Done c135910db0b2b1bc4488ce8d026a575a70e0001f